functions/accept.js

```diff
@@ -1,44 +1,169 @@
 "use strict";
 
+const Hawk = require("hawk");
+const Busboy = require("busboy");
 const AWS = require("aws-sdk");
 const S3 = new AWS.S3({ apiVersion: "2006-03-01" });
 const SQS = new AWS.SQS({ apiVersion: "2012-11-05" });
+const documentClient = new AWS.DynamoDB.DocumentClient();
+const { DEV_CREDENTIALS, DEFAULT_HAWK_ALGORITHM } = require("../lib/constants");
 
-const { QUEUE_NAME, CONTENT_BUCKET } = process.env;
-module.exports.handler = async function(
-  { requestContext: { requestId } },
-  context
-) {
-  console.time("accept");
-  const responseCode = 200;
-  const responseBody = { requestId };
-
-  console.time("acceptS3");
-  const result = await S3.putObject({
-    Bucket: CONTENT_BUCKET,
-    Key: requestId,
-    Body: "THIS WILL BE AN IMAGE SOMEDAY"
-  }).promise();
-  responseBody.s3Result = result;
-  console.timeEnd("acceptS3");
+const REQUIRED_FIELDS = ["image", "negative_uri", "positive_uri"];
+
+module.exports.post = async function(event, context) {
+  const { QUEUE_NAME: QueueName, CONTENT_BUCKET: Bucket } = process.env;
+
+  const {
+    headers,
+    queryStringParameters: params,
+    requestContext: { path, requestId }
+  } = event;
+
+  const {
+    Host: host,
+    Authorization: authorization,
+    "X-Forwarded-Port": port = 80
+  } = headers;
+
+  let credentials;
+  try {
+    ({ credentials } = await Hawk.server.authenticate(
+      {
+        method: "POST",
+        url: path,
+        params,
+        host,
+        port,
+        authorization
+      },
+      lookupCredentials
+    ));
+  } catch (err) {
+    return response(
+      401,
+      { error: err.message },
+      { "WWW-Authenticate": "Hawk" }
+    );
+  }
 
-  console.time("acceptSQS");
-  const { QueueUrl } = await SQS.getQueueUrl({
-    QueueName: QUEUE_NAME
+  let body, negative_uri, positive_uri, positive_email, notes, image;
+  try {
+    body = await parseRequestBody(event);
+    REQUIRED_FIELDS.forEach(name => {
+      if (!body[name]) {
+        throw { message: `Required "${name}" is missing` };
+      }
+    });
+    // TODO: More input validation here?
+    ({ negative_uri, positive_uri, positive_email, notes, image } = body);
+  } catch (err) {
+    return response(400, { error: err.message });
+  }
+
+  const imageKey = `image-${requestId}`;
```

Comment:
This also seems like a weird quirk: I had to prefix the `requestId` with something (e.g. `image-`) or else the object would never show up in S3. No errors, just nothing in the bucket.